app/utils/feed_discovery.rb

```diff
@@ -1,20 +1,22 @@
 require "feedbag"
 require "feedjira"
+require "httparty"
 
 class FeedDiscovery
-  def discover(url, finder = Feedbag, parser = Feedjira::Feed)
-    get_feed_for_url(url, parser) do
+  def discover(url, finder = Feedbag, parser = Feedjira, client = HTTParty)
+    get_feed_for_url(url, parser, client) do
       urls = finder.find(url)
       return false if urls.empty?
 
-      get_feed_for_url(urls.first, parser) do
+      get_feed_for_url(urls.first, parser, client) do
         return false
       end
     end
   end
 
-  def get_feed_for_url(url, parser)
-    feed = parser.fetch_and_parse(url)
+  def get_feed_for_url(url, parser, client)
```

Comment:
Hi @mikeastock , I looked into it a little more and it seems that some feeds are no longer working. I tracked the issue to this line. I tried it on racksh. The following doesn't work:

```
FeedDiscovery.new.get_feed_for_url('http://mkremins.github.io/atom.xml', Feedjira, HTTParty)
```

It returns nil. While the previous code was working:

```
Feedjira::Feed.fetch_and_parse('http://mkremins.github.io/atom.xml')
```

Do you know what's happening?

Reply:
I'll dig into this this weekend and get back to you.

Reply:
@mikeastock it appears the issue was that HTTParty.get returns a response, not a string body. Adding two `.to_s` fixed things for me

Reply:
@dwaite I'd like to merge this to my fork at least seeing that nothing seems to be merged on mainline anymore. May have to migrate to another reader eventually. Was the fix you tested out appending the `.to_s` to the following 2 lines? Thanks.